src/core/tiledmesh/qgstiledmeshlayerrenderer.cpp

```diff
@@ -0,0 +1,72 @@
+/***************************************************************************
+                         qgstiledmeshlayerrenderer.cpp
+                         --------------------
+    begin                : June 2023
+    copyright            : (C) 2023 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "qgstiledmeshlayerrenderer.h"
+#include "qgstiledmeshlayer.h"
+#include "qgsfeedback.h"
+#include "qgsmapclippingutils.h"
+#include "qgsrendercontext.h"
+
+QgsTiledMeshLayerRenderer::QgsTiledMeshLayerRenderer( QgsTiledMeshLayer *layer, QgsRenderContext &context )
+  : QgsMapLayerRenderer( layer->id(), &context )
+  , mFeedback( new QgsFeedback )
+{
+  // We must not keep pointer to mLayer (it's dangerous) - we must copy anything we need for rendering
+  // or use some locking to prevent read/write from multiple threads
+  if ( !layer->dataProvider() ) // || !layer->renderer() )
```

Comment:
Is there something to clean here before merging?

Reply:
No, it'll get fixed when the renderer itself is implemented